### ISSUE TITLE
CI: fix `check_storage_layout` for new contracts

### ISSUE THREAD

**Issue by rndquu** [ubiquity/uad-ui#972]
There's the [core-contracts-storage-check](https://github.com/ubiquity/ubiquity-dollar/blob/20799bedf5ee4d40d2863653f7bc2fe68a2ce3fa/.github/workflows/core-contracts-storage-check.yml) workflow which checks that contracts' storage is not messed on contract upgrades.

Check these CI runs which fails with the `No workflow run found with an artifact named...` error:
- https://github.com/ubiquity/ubiquity-dollar/actions/runs/11125075713/job/30912062971?pr=959
- https://github.com/ubiquity/ubiquity-dollar/actions/runs/11145695030/job/30975996318?pr=971

The issue is that when a **new contract** is added the [core-contracts-storage-check](https://github.com/ubiquity/ubiquity-dollar/blob/20799bedf5ee4d40d2863653f7bc2fe68a2ce3fa/.github/workflows/core-contracts-storage-check.yml) starts to fail while the expected behavior for the CI workflow to pass because we don't need to check storage layout for newly added contracts (because storage of the newly added contract hasn't been updated hence storage collision could not occur).

So as a part of this issue we should make sure that:
1. [core-contracts-storage-check](https://github.com/ubiquity/ubiquity-dollar/blob/20799bedf5ee4d40d2863653f7bc2fe68a2ce3fa/.github/workflows/core-contracts-storage-check.yml) is not failing for newly added contracts
2. [diamond-storage-check.yml](https://github.com/ubiquity/ubiquity-dollar/blob/20799bedf5ee4d40d2863653f7bc2fe68a2ce3fa/.github/workflows/diamond-storage-check.yml) is not failing for newly added libraries

P.S. Pls provide a QA in the related PR with these scenarios:
1) No storage updates, CI passing
2) Storage update, no collision, CI passing
3) Storage update, collision, CI failing
4) New contract added, CI passing

**Comment by zugdev** [comment]
Can use [tj-actions/changed-files](https://github.com/tj-actions/changed-files) for this.

**Comment by rndquu** [comment]
> Can use [tj-actions/changed-files](https://github.com/tj-actions/changed-files) for this.

https://github.com/ubiquity/ubiquity-dollar/blob/20799bedf5ee4d40d2863653f7bc2fe68a2ce3fa/.github/workflows/core-contracts-storage-check.yml#L25

**Comment by rndquu** [comment]
I suppose the simplest solution would be to add a new job step [here](https://github.com/ubiquity/ubiquity-dollar/blob/20799bedf5ee4d40d2863653f7bc2fe68a2ce3fa/.github/workflows/core-contracts-storage-check.yml#L30) and [here](https://github.com/ubiquity/ubiquity-dollar/blob/20799bedf5ee4d40d2863653f7bc2fe68a2ce3fa/.github/workflows/diamond-storage-check.yml#L34) to filter modifed contracts that exist both in the PR and default branches.

**Comment by zugdev** [comment]
Yes already got it. I meant using tj-actions to see if contract existed last commit in whatever branch.

**Comment by zugdev** [comment]
[Successful contract skip run on push](https://github.com/zugdev/ubiquity-dollar/actions/runs/11171039998/job/31055032949)

**Comment by zugdev** [comment]
> 1. No storage updates, CI passing
[here](https://github.com/zugdev/ubiquity-dollar/actions/runs/11171125690)

> 2. Storage update, no collision, CI passing
[here](https://github.com/zugdev/ubiquity-dollar/actions/runs/11171098048/job/31055202899)

> 3. Storage update, collision, CI failing
in a sec

> 4. New contract added, CI passing
[here](https://github.com/zugdev/ubiquity-dollar/actions/runs/11171039998)

**Comment by zugdev** [comment]
I should be reassigned here, waiting for review.

**Comment by molecula451** [comment]
i mean i asked for rndqnuu to review PRs to see if its already done or not, so it can be work out if it's not, my guess

**Comment by saimuren** [comment]
I'd like to fix this. I've already been working on the `foundry-storage-check` action (see PR ubiquity/foundry-storage-check#3) for the related issue #992.

### Root Cause Analysis
The `core-contracts-storage-check` workflow fails for new contracts because:
1. New contracts don't have a baseline storage layout to compare against
2. The action expects a pre-existing layout artifact

### Proposed Fix
1. Skip storage check for contracts that don't have a baseline layout
2. Auto-generate baseline layout on first run for new contracts
3. Update the workflow to handle the "new contract" case gracefully

Since I already have context on the `foundry-storage-check` action internals, I can implement this efficiently. Ready to start immediately.

**Comment by lustsazeus-lab** [comment]
Claiming this bounty from lustsazeus-lab. I’ll implement a CI fix so storage-layout checks pass for newly added contracts/libraries while preserving collision detection behavior for actual storage changes.

**Comment by lustsazeus-lab** [comment]
Implemented in PR: https://github.com/ubiquity/ubiquity-dollar/pull/1008

### QA matrix
Ran local deterministic validation:
```bash
bash .github/scripts/test-build-storage-check-matrix.sh
```

| Scenario | Result |
|---|---|
| 1) No storage updates | ✅ Pass (`[]` matrix) |
| 2) Storage update, no collision | ✅ Pass (existing file included in matrix, storage check still runs) |
| 3) Storage update, collision | ✅ Expected fail path preserved (existing file included in matrix, collision detection still runs downstream) |
| 4) New contract/library added | ✅ Pass (new files skipped from matrix when absent on base sha, preventing missing-baseline-artifact failures) |

Validation output excerpt:
```
Skipping new file without baseline on base sha: packages/contracts/src/dollar/core/NewCore.sol
Skipping new file without baseline on base sha: packages/contracts/src/dollar/libraries/LibNew.sol
All storage-matrix QA scenarios passed.
```

**Comment by lustsazeus-lab** [comment]
Implemented and opened PR: https://github.com/ubiquity/ubiquity-dollar/pull/1009

This updates both storage-check workflows to skip newly-added contracts/libraries on PRs (where no base artifact exists yet), while preserving existing storage collision checks for pre-existing files.

**Comment by 360john360** [comment]
Hi, I had a look at this and can work on a focused fix locally. I noticed the bot responses say this task requires a core team/admin/collaborator role to start. Before I open a PR, could you confirm whether this issue is currently open to external contributors and whether an external contributor can be assigned/paid for it?

**Comment by sandliu** [comment]
I’d like to work on this. I’ll submit a PR with the requested QA scenarios.

**Comment by sandliu** [comment]
@rndquu I opened PR #1038 with a focused workflow fix for this issue: https://github.com/ubiquity/ubiquity-dollar/pull/1038

The PR was automatically closed by ubiquity-os after creation. The latest bot response here says this task currently requires a collaborator/core team/admin role to start, while my role is contributor.

Could you please confirm whether this issue is open to external contributors? If yes, could you manually allow/assign me and reopen PR #1038, or advise the expected process? The branch is still available at sandliu:storage-check-added-contracts, and CodeRabbit reported no actionable comments.

**Comment by 8hatay9-commits** [comment]
@rndquu @0x4007 I’m ready to implement this and provide the requested QA scenarios. `/start` is currently rejecting my account because the task only allows collaborator/core roles. If this issue is still active, could you authorize external contributors (or assign `8hatay9-commits`) so I can start immediately?

**Comment by rushikeshgarad2024-dev** [comment]
I've submitted a fix for this: **PR #1059** — https://github.com/ubiquity/ubiquity-dollar/pull/1059

### What was fixed

Both `core-contracts-storage-check.yml` and `diamond-storage-check.yml` now use the `contracts_added_files` / `libraries_added_files` outputs from `tj-actions/changed-files@v42` to detect brand-new files and exclude them from the matrix before the storage-layout diff job runs.

This means:
- **Newly added contracts/libraries** → skipped (no prior artifact exists, no collision possible) ✅
- **Modified existing contracts/libraries** → still go through the full storage-layout diff ✅

All 4 QA scenarios from the issue description are documented in the PR.
